File: Aman_Bhaiya_DSA_List/Arrays/LQ23.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        out = []
        out.append(nums[0])
        n = len(nums)
        for i in range(1,n):
            out.append(out[i-1] * nums[i])

        rp = 1
        for i in range(n-1,0,-1):
            out[i] = out[i-1] * rp
            rp = rp * nums[i]
        
        out[0] = rp
        return out

File: Aman_Bhaiya_DSA_List/Arrays/LQ23.java
public class LQ23 {
    public int[] productExceptSelf(int[] nums) {
        int out[] = new int[nums.length];
        out[0] = nums[0];
        for(int i = 1; i < nums.length; i++){
            out[i] = out[i-1]*nums[i];
        }
        
        int rp = 1;
        for(int i = nums.length - 1; i > 0; i--){
            out[i] = out[i-1] * rp;
            rp *= nums[i];
        }
        out[0] = rp;
        return out;
    }
}
